public/js/length_visualisation.js

```diff
@@ -0,0 +1,339 @@
+import _ from 'underscore';
+import VisualisationHelpers from './visualisation_helpers';
+
+export default class LengthVisualisation {
+  constructor(query, svg_container, algorithm) {
+    this._svg_container_d3 =  d3.select(svg_container[0]);
+    this._svg = {};
+    this._query = query;
```

Comment:
`_` is used to indicate that this property is private. We don't need to do that for query.